### PR TITLE
contrib/gdm-settings: new package (4.4)

### PR DESCRIPTION
**Pull request by Gnarwhal**

### CODE REVIEW

File: user/gdm-settings/template.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+build_style = "meson"
+hostmakedepends = [
+    "blueprint-compiler",
+    "meson",
+    "gobject-introspection",
+]
+makedepends = [
+    "appstream-glib",
+    "desktop-file-utils",
+    "gettext",
```
Comment (nekopsykose) [inline user/gdm-settings/template.py:13]: hostmake

File: user/gdm-settings/template.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    "glib-devel",
+    "gtk4-devel",
+    "libadwaita-devel",
+    "python-gobject-devel",
+    "python-pkgconfig",
```
Comment (nekopsykose) [inline user/gdm-settings/template.py:18]: host

**Comment by nekopsykose** [comment]
put it in contrib/

File: user/gdm-settings/template.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+]
+depends = [
+    "gdm",
+]
+pkgdesc = "Settings app for GDM"
+maintainer = "Gnarwhal <git.aspect893@passmail.net>"
+license = "AGPL-3.0-or-later"
+url = "https://github.com/gdm-settings/gdm-settings"
+source = f"https://github.com/gdm-settings/gdm-settings/archive/refs/tags/v{pkgver}.tar.gz"
```
Comment (nekopsykose) [inline user/gdm-settings/template.py:27]:
```suggestion
source = f"{url}/gdm-settings/archive/refs/tags/v{pkgver}.tar.gz"
```

File: user/gdm-settings/template.py (hunk 2 above)
Comment (nekopsykose) [reply user/gdm-settings/template.py:18]: though i think this is not used? you only want regular `pkgconf` afaict, could be wrong just had a quick look

**Comment by nekopsykose** [comment]
i guess the workaround works but this code doesn't make any sense to me. what is the point of calling `cp` `mkdir` `rm` from python? the stdlib has versions of all of this..

e.g. `cp -r a -t b` can be `shutil.copy_tree(a, b)` (they both copy a tree recursively and fail when `b` exists). i would rather just see that fixed upstream instead of shell workarounds, but it would take quite a bit of refactoring around CommandList to also invoke a passed function instead of only `Command('sh', ...`

**Comment by Gnarwhal** [comment]
My only guess for why it is done this way is so that gdm-settings can be run without root and then use pkexec to apply changes as root? But I don't know python well enough to know if there is a better work around ¯\_(ツ)_/¯

**Comment by nekopsykose** [comment]
i guess that's a somewhat interesting reason that makes it (even) more work to do something else with

**Comment by Gnarwhal** [comment]
New proposal: what if there was a supplementary program (call it `gdm-settings-applicator`) pre-written. And instead of adding shell commands directly to a list that get written to a temporary file and `sh`ed, settings.py setup command line arguments for the applicator and then the applicator as a whole was `pkexec`ed.

**Comment by nekopsykose** [comment]
sure, ask upstream :D

**Comment by nekopsykose** [comment]
that said this is fine for now :) just nice if made better later

**Pull request merged**